giskard/scanner/robustness/numerical_perturbation_detector.py

```diff
@@ -0,0 +1,28 @@
+from typing import Sequence
+
+import numpy as np
+
+from ...datasets.base import Dataset
+from ...models.base import BaseModel
+from ..decorators import detector
+from .base_numerical_detector import BaseNumericalPerturbationDetector
+
+
+@detector(
+    name="numerical_perturbation",
+    tags=[
+        "numerical_perturbation",
+        "robustness",
+        "classification",
+        "regression",
+    ],
+)
+class NumericalPerturbationDetector(BaseNumericalPerturbationDetector):
+    """Detects robustness problems in a model by applying numerical perturbations to the numerical features."""
+
+    def _get_default_perturbations(self, model: BaseModel, dataset: Dataset) -> Sequence:
+        return [
+            lambda x: x * 1.01,
+            lambda x: x * 0.99,
+            lambda x: x + np.random.normal(0, 0.01, x.shape),
```

Comment:
These should be proper transformation functions (extending `TransformationFunction`).

IMPORTANT: all of these perturbations assume that the numerical values are floats. This is not always the case, integer features are common: when such data types are present we must not alter that. Applying the transformations above (for example Gaussian noise) to an integer feature like `num_passengers` would silently convert the data type to float, potentially breaking the model inference.

Reply:
So how do you suggest I check perturbation for integer data?